### PR TITLE
RAZ Implementation for Azure

### PR DESCRIPTION
**Pull request by cmperro**
This PR adds RAZ support for Azure in cloudera.exe.

- Adds a switch for enabling RAZ (bool)
- Creates a new RAZ MSI and assigns the necessary Roles to the MSI
- Updates the idbroker mapping task to pass the cloud authorizer role (RAZ)
- Updates the datalake creation task to pass the enableRAZ boolean

This PR needs to be tested in conjunction with [cloudera.cloud PR 55](https://github.com/cloudera-labs/cloudera.cloud/pull/55)

### CODE REVIEW

File: roles/common/defaults/main.yml [hunk 1]
```diff
@@ -45,6 +45,7 @@ common__data_suffix:                      "{{ globals.labels.data | default('dat
 common__external_data_suffix:             "{{ globals.labels.external_data | default('external') }}"
 common__datalake_admin_suffix:            "{{ globals.labels.datalake_admin | default('dladmin') }}"
 common__ranger_audit_suffix:              "{{ globals.labels.ranger_audit | default('audit') }}"
+common__raz_suffix:                       "{{ globals.labels.raz | default('raz') }}"
```
Comment (wmudge) [inline roles/common/defaults/main.yml:48]: Document in `configuration.yml`

File: roles/platform/defaults/main.yml [hunk 2]
```diff
@@ -77,6 +77,7 @@ plat__xacccount_credential_name:              "{{ common__xaccount_credential_na
 plat__workload_analytics:                     "{{ env.workload_analytics | default(True) }}"
 plat__tunnel:                                 "{{ common__tunnel }}"
 plat__public_endpoint_access:                 "{{ common__public_endpoint_access }}"
+plat__enable_raz:                             "{{ env.datalake.enable_raz | default(False) }}"
```
Comment (wmudge) [inline roles/platform/defaults/main.yml:80]: Document in `configuration.yml`

File: roles/platform/defaults/main.yml [hunk 3]
```diff
@@ -177,6 +178,7 @@ plat__azure_data_suffix:                      "{{ env.azure.role.label.data | de
 plat__azure_identity_suffix:                  "{{ env.azure.role.label.identity | default(common__identity_suffix) }}"
 plat__azure_datalake_admin_suffix:            "{{ env.azure.role.label.datalake_admin | default(common__datalake_admin_suffix) }}"
 plat__azure_ranger_audit_suffix:              "{{ env.azure.role.label.ranger_audit | default(common__ranger_audit_suffix) }}"
+plat__azure_raz_suffix:                       "{{ env.azure.role.label.raz | default(common__raz_suffix) }}"
```
Comment (wmudge) [inline roles/platform/defaults/main.yml:181]: Docs

File: roles/platform/defaults/main.yml [hunk 4]
```diff
@@ -200,6 +202,7 @@ plat__azure_policy_url:                       "{{ env.azure.policy.url | default
 plat__azure_log_identity_name:                "{{ env.azure.role.name.log | default([plat__namespace, plat__azure_log_suffix, plat__azure_identity_suffix] | join('-')) }}"
 plat__azure_datalakeadmin_identity_name:      "{{ env.azure.role.name.datalake_admin | default([plat__namespace, plat__azure_datalake_admin_suffix, plat__azure_identity_suffix] | join('-')) }}"
 plat__azure_ranger_audit_identity_name:       "{{ env.azure.role.name.ranger_audit | default([plat__namespace, plat__azure_ranger_audit_suffix, plat__azure_identity_suffix] | join('-')) }}"
+plat__azure_raz_identity_name:                "{{ env.azure.role.name.raz | default([plat__namespace, plat__azure_raz_suffix, plat__azure_identity_suffix] | join('-')) }}"
```
Comment (wmudge) [inline roles/platform/defaults/main.yml:205]: Docs

File: roles/platform/tasks/initialize_teardown_azure.yml [hunk 5]
```diff
@@ -40,5 +40,5 @@
       - "{{ plat__azure_datalakeadmin_identity_name }}"
       - "{{ plat__azure_log_identity_name }}"
       - "{{ plat__azure_ranger_audit_identity_name }}"
-    role_assignment_assignee_list: "{{ msi_principal_id_list | union([plat__azure_application_service_principal_objuuid]) | list }}"
+    role_assignment_assignee_list: "{{ msi_principal_id_list | union([plat__azure_application_service_principal_objuuid | default([])]) | list }}"
```
Comment (wmudge) [inline roles/platform/tasks/initialize_teardown_azure.yml:43]: Does this return a nested list if `plat__azure_application_service_principal_objuuid` is undef?

File: roles/platform/tasks/setup_azure_authz.yml [hunk 6]
```diff
@@ -136,11 +133,7 @@
   register: __azure_identity_list
   delay: 5
   retries: 120 # 10 mins
-  until:
-    - plat__azure_idbroker_identity_name in discovered_msi_list
-    - plat__azure_datalakeadmin_identity_name in discovered_msi_list
-    - plat__azure_log_identity_name in discovered_msi_list
-    - plat__azure_ranger_audit_identity_name in discovered_msi_list
+  until: discovered_msi_list is subset(plat__azure_msis)
```
Comment (wmudge) [inline roles/platform/tasks/setup_azure_authz.yml:136]: +1!

**Review by wmudge — CHANGES_REQUESTED** [review]
Minor changes. Needs a rebase as well.

**Review by Chaffelson — CHANGES_REQUESTED** [review]
Still needs rebase

**Comment by cmperro** [comment]
> 

Addressed requests and rebased.

**Review by wmudge — APPROVED** [review]

**Comment by Chaffelson** [comment]
@cmperro @wmudge Can we revisit this PR please to see if we should complete it for the next release, or if it has been superceded?

**Comment by wmudge** [comment]
@cmperro This needs a rebase on `devel` again, post-CDW PR merge.

**Review by wmudge — APPROVED** [review]

**Pull request merged**